docs/migrations-guidelines.md

```diff
@@ -0,0 +1,89 @@
+# Migration and Testing Guidelines
+
+## Overview
+
+This document outlines best practices and guidelines for writing migration scripts and their corresponding test cases.
+
+The focus is on ensuring data integrity, handling errors gracefully, and maintaining consistency across mobile app versions.
+
+We always look for improvement, if you see anything that could be improved, please open a PR against this guidelines.
+
+You can also check an example of a migration on MetaMask mobile app [here](https://github.com/MetaMask/metamask-mobile/blob/1855bd674e33bb0ece06fb6d8f09a4e5df46a108/app/store/migrations/044.ts#L1)
+
+## Migration Guidelines
+
+1. **Pre-Validation Checks**:
+
+- Ensure the state to be migrated is valid before running the migration logic,
+- Verify the version and integrity of the state with utility functions like `ensureValidState`.
+
+2. **Error Handling**:
+
+The following prevents manipulation of potentially corrupted data, enforcing corrective action before proceeding.
```

Comment:
This sentence seems to be discussing the "state integrity checks". Maybe we can swap the places of these two guidelines? So that we start by explaining the integrity checks, then follow with explaining how to handle errors resulting from them.